### PR TITLE
Fix a file creation check.

### PR DESCRIPTION
**Pull request by drwells**
Commit 1cdbeecc858 is broken when step-42 is run with multiple MPI processes.

### CODE REVIEW

File: examples/step-42/step-42.cc [hunk 1]
```diff
@@ -876,8 +876,15 @@ namespace Step42
...
       output_dir += "/";
-    const int ierr = mkdir(output_dir.c_str(), 0777);
-    AssertThrow(ierr == 0, ExcIO());
+
+    if (Utilities::MPI::this_mpi_process(mpi_communicator) == 0)
+      {
+        const int ierr = mkdir(output_dir.c_str(), 0777);
+        AssertThrow(ierr == 0, ExcIO());
+      }
+    const int ierr = MPI_Barrier(mpi_communicator);
```
Comment (Rombur) [inline examples/step-42/step-42.cc]: Why did you add the `MPI_Barrier`?

Comment (drwells) [reply examples/step-42/step-42.cc]: I thought that I needed it but I don't recall why. I got rid of it.

File: examples/step-42/step-42.cc [hunk 2]
```diff
@@ -876,8 +876,12 @@ namespace Step42
     output_dir = prm.get("output directory");
     if (output_dir != "" && *(output_dir.rbegin()) != '/')
       output_dir += "/";
-    const int ierr = mkdir(output_dir.c_str(), 0777);
-    AssertThrow(ierr == 0, ExcIO());
+
+    if (Utilities::MPI::this_mpi_process(mpi_communicator) == 0)
+      {
+        const int ierr = mkdir(output_dir.c_str(), 0777);
+        AssertThrow(ierr == 0, ExcIO());
```
Comment (masterleinad) [inline examples/step-42/step-42.cc]: Do we want to allow that the directory already exists?

Comment (drwells) [reply examples/step-42/step-42.cc]: Good point. I added a check for this.
